posts/Modernize-your-pipeline-to-be-ARM-ready-with-multi-arch-container-builds.md

```diff
@@ -0,0 +1,233 @@
+
+## Cloud Device Pool and Cost Savings
+
+The term cloud agnostic means being able to run your application on any cloud provider. The same goes for architecture. Architecture agnosticism means a piece of software can run on any cpu architecture (within reason).  Some examples include x86 (32 bit), arm64 (sometimes called aarch64), and amd64 (also known as x86_64).  These are by far the most popular choices.  Again, building multi-arch containers increases the number of devices that piece of software can run on.  Not only that, but certain workloads work better on different cpu architectures, being more power efficient and can save you money!
+
+One final use case example before we dive into the how.  IOT applications have to run on resource constrained Single Board Computers.  This is especially true for greenfield IOT development where many different test devices may be used.  Having multi-arch images ready to go means being able to test across multiple devices.  It can even help keep a company alive.  If you had specifically built a piece of software to run on a Raspberry PI (ARM64), your company may have gone out of business during the pandemic when the things became impossible to find! 
+
+# Where this work happens
+
+## Either way, it's turtles all the way down<sup>1</sup>
+
+It may be confusing at first to understand exactly what it means to convert your existing workloads to be able to run on multiple architectures. Let us look at a gross oversimplification.... All code inevitably ends up being ran as machine code that uses a given CPU's instruction set. In other words, it's all 1s and 0s eventually. For x86_64 and ARM, look at the machine code that is running for each architecture and it will be different.
+
+The cool part about this whole thing is, changes in your application code may not even be required. Often times it is as easy as using a different compiler flag, running interpreted code with an interpreter compiled for a different architecture, or instructing `docker buildx` to build an image for a different architecutre. That's right, these changes are made during a phase of the software development lifecycle called the release phase, specifically the "build process". For this article, we will be focused on a build process that takes place in a build pipeline, namely, gitlab pipelines. 
```

Comment:
This sentence feels too wordy

That's right, these changes are made during a phase of the software development lifecycle called the release phase, specifically the "build process".

Reply:
Is the build phase part of the release phase?